src/renderer/vt/state.cpp

```diff
@@ -527,3 +527,10 @@ HRESULT VtEngine::RequestWin32Input() noexcept
     RETURN_IF_FAILED(_Flush());
     return S_OK;
 }
+
+HRESULT VtEngine::SwitchScreenBuffer(const bool useAltBuffer) noexcept
+{
+    RETURN_IF_FAILED(_SwitchScreenBuffer(useAltBuffer));
+    RETURN_IF_FAILED(_Flush());

```

Comment:
order of operations -- interesting. Can this cause torn state?

Reply:
Hmm. So when we switch to the alt buffer, we first
* flush out the current frame
* emit this sequence, and flush again. 

so I think this is basically atomic, and then we return back out and continue processing more output.